cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/configs/StorageProperties.java

```diff
@@ -0,0 +1,35 @@
+package com.linkedin.openhouse.cluster.storage.configs;
+
+import com.linkedin.openhouse.cluster.configs.YamlPropertySourceFactory;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.*;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@ConfigurationProperties(prefix = "cluster.storages")
+@PropertySource(
+    name = "clusterStorage",
+    value = "file:${OPENHOUSE_CLUSTER_CONFIG_PATH:/var/config/cluster.yaml}",
+    factory = YamlPropertySourceFactory.class,
+    ignoreResourceNotFound = true)
```

Comment:
is this a safe config?

Reply:
what aspect of safe were you thinking of? its safe in the sense, if properties are not provided, spring-boot will not fail and storageProperties will be null.

Reply:
that's exactly the thing I am thinking. Should we allow application to boot if storageProperties is null ?

Reply:
Got it! I'd like to follow previous pattern of `if not provided null/default` and let the beans being initialized fail with useful error message.